lib/state_machine/transition.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include "message.cpp"
+#include "state.hpp"
+
+namespace hyped::state_machine {
+
+// Transition class.
+// Each transition has a current state and next state, as well as a message indicating the kind of
+// transition.
+class Transition {
+ public:
+  Transition(State from_state, State to_state, Message transition_message)
```

Comment:
Should this not be a separate function within this class? Also why are we storing all of this as class members?